### PR TITLE
docs: add release notes for 0.16.0

### PR DESCRIPTION
**Pull request by dzhu**
Add release notes for 0.16.0.

### CODE REVIEW

File: docs/release-notes.txt [hunk 1]
```diff
@@ -5,9 +5,92 @@
 ###############
 
 **************
- Version 0.15
+ Version 0.16
```
Comment (justin-determined-ai) [inline docs/release-notes.txt:8]: blocking: can you retain the 0.15 header below for all the 0.15 changes?

File: docs/release-notes.txt [hunk 2]
```diff
@@ -5,9 +5,92 @@
...
+==============
+
+**Release Date:** June 14, 2021
+
+**New Features**
+
+-  Python SDK: Extend the :ref:`Checkpoint Export API
+   <exporting-checkpoints>` into a Python SDK capable of launching and
+   controlling experiments on the cluster directly from Python. See the
+   docs and examples in the :mod:`~determined.experimental.client`
```
Comment (justin-determined-ai) [inline docs/release-notes.txt:21]: non-blocking: "documentation" (rather than "docs")?

File: docs/release-notes.txt [hunk 3]
```diff
@@ -5,9 +5,92 @@
...
+
+-  REST API: Add new parameters to ``/api/v1/experiments`` to filter and
+   sort experiments by name.
+
+-  Master configuration: Support ``bind_mounts`` in
+   ``task_container_defaults`` in the master configuration. The
+   configured directories will be mounted for experiments, notebooks,
+   commands, shells, and TensorBoards.
+
+-  Images: Add an environment image containing TensorFlow 2.5 and CUDA
```
Comment (justin-determined-ai) [inline docs/release-notes.txt:43]: non-blocking: should we link to the environments repo for this?

**Review by justin-determined-ai — APPROVED** [review]
LGTM, just one change and some thoughts. Thanks for putting this together!

File: docs/release-notes.txt (hunk 1 above)
Comment (dzhu) [reply docs/release-notes.txt:8]: Oops, of course!

File: docs/release-notes.txt (hunk 2 above)
Comment (dzhu) [reply docs/release-notes.txt:21]: Sure!

File: docs/release-notes.txt (hunk 3 above)
Comment (dzhu) [reply docs/release-notes.txt:43]: Eh, it doesn't look like the precedent is to have have any links for new images.

**Pull request merged**